### PR TITLE
[model] Handle loss layer to be added from user

### LINKED ISSUE #744: Allow loss layer to be created and added by user
Allow loss layer to be created and added by the user.

### PR DESCRIPTION
**Pull request by kparichay**
Handle loss layer to be added from the user which is created with c++ API

Resolves #744

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: nntrainer/models/neuralnet.cpp [hunk 1]
```diff
@@ -186,9 +188,22 @@ int NeuralNetwork::init() {
...
+  if (initialized) {
+    ml_loge("Error: Initializing the model again");
+    return ML_ERROR_NOT_SUPPORTED;
+  }
+
   status = isInitializable();
   NN_RETURN_STATUS();
 
+  /** Add loss layer if not already added */
+  if (layers.back()->getType() != LayerType::LAYER_LOSS) {
```
Comment (zhoonit) [inline nntrainer/models/neuralnet.cpp:200]: How about add a warning if the user is overriding the loss set to the model?

**Review by zhoonit — APPROVED** [review]
LGTM :)

File: nntrainer/models/neuralnet.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/models/neuralnet.cpp:200]: done.

**Pull request merged**